plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/internal/actions/DefaultActionDispatcher.java

```diff
@@ -87,11 +87,23 @@ public class DefaultActionDispatcher extends Disposable implements ActionDispatc
    protected Provider<GLSPClient> client;
 
    public DefaultActionDispatcher() {
+      this.initialize();
+   }
+
+   protected void initialize() {
       this.name = getClass().getSimpleName() + " " + COUNT.incrementAndGet();
-      this.thread = new Thread(this::runThread);
-      this.thread.setName(this.name);
-      this.thread.setDaemon(true);
-      this.thread.start();
+      this.thread = this.createThread();
```

Comment:
This throws a compile error. You probably need to remove the final declaration of thread.

Reply:
Ah, sorry, I have issues compiling it locally so I was doing it blindly, hopefully this will fix it, thank you!